### DISCUSSION TITLE
Customize `email` and `name` used to send email with Mail::to()

### DISCUSSION THREAD

**Discussion by ProgrammerZ** [laravel/framework#52981] [Q&A] [answered]
I would like to send email with `Mail::to($user)` Facade, but can't figure our how to customize attributes that is used for email and name when using Model. I can't use `email` or `name` attributes on the user Model and I can't create custom Attributes for it as well. 
Is there anything similar to `routeNotificationForMail()` but just for Mail?

**Comment by shaedrich** [comment]
Being able to set the receipient but not the sender is something that probably confuses a lot of Laravel developers that have the opposite user case of the one who initially implemented it. If [using the global `config('mail.from')`](https://laravel.com/docs/10.x/mail#using-a-global-from-address) does not work for you, you have a couple of options:

If you use mailables, you can [use an envelope](https://laravel.com/docs/10.x/mail#using-the-envelope) to achieve that:
```php
use Illuminate\Mail\Mailables\Address;
use Illuminate\Mail\Mailables\Envelope;
 
/**
 * Get the message envelope.
 */
public function envelope(): Envelope
{
    return new Envelope(
        from: new Address('jeffrey@example.com', 'Jeffrey Way'),
        subject: 'Order Shipped',
    );
}
```

If you use notifications, you can [customize the `MailMessage`'s sender](https://laravel.com/docs/10.x/notifications#customizing-the-sender) via the notification's `toMail()` method:
```php
/**
 * Get the mail representation of the notification.
 */
public function toMail(object $notifiable): MailMessage
{
    return (new MailMessage)
                ->from('barrett@example.com', 'Barrett Blair')
                ->line('...');
}
```

**Reply by ProgrammerZ** [reply, marked as answer]
Thanks, I was probably not clear enough with my question. My goal was to fetch recipient address bases on User Model attributes, not per Mailable, but I could not use `name`  or `email` on the User Model.
In short you can't do that with `Mail`, so instead you can use `Notification` with `routeNotificationForMail()` on User model.
